src/lib/utils.js

```diff
@@ -488,6 +490,34 @@ const validateActorName = (actorName) => {
     }
 };
 
+const getPythonCommand = (directory) => {
+    const pythonVenvPath = /^win/.test(process.platform)
+        ? './.venv/bin/python3.exe'
+        : './.venv/bin/python3';
+
+    if (process.env.VIRTUAL_ENV) {
+        directory = process.env.VIRTUAL_ENV;
+    }
+    const fullPythonVenvPath = path.join(directory, pythonVenvPath);
+    if (fs.existsSync(fullPythonVenvPath)) {
+        return pythonVenvPath;
+    }
+    return 'python3';
+};
+
+const detectPythonVersion = (directory) => {
+    const pythonCommand = getPythonCommand(directory);
+    try {
+        return execSync(`${pythonCommand} -c "import platform; print(platform.python_version(), end='')"`, { encoding: 'utf-8' });
+    } catch {
+        return undefined;
+    }
+};
+
+const isPythonVersionSupported = (installedPythonVersion) => {
+    return semver.gte(installedPythonVersion, MINIMUM_SUPPORTED_PYTHON_VERSION);
```

Comment:
purely theoretically, it's better to do `.satisfies(installed, "^" + MINIMUM)` - so that it doesn't match on Python4 :)

Reply:
I don't think there will ever be a Python 4, after the Python 2 -> 3 pains, but one can always hope 😄